src/core/astParser.ts

```
@@ -0,0 +1,147 @@
+
+    // Extract symbols
+    try {
+      const symbols = languageConfig.extractSymbols(tree, sourceCode);
+      return symbols;
+    } catch (error) {
+      console.warn(`Failed to extract symbols from ${filePath}:`, error instanceof Error ? error.message : 'Unknown error');
+      return [];
+    }
+  }
+
+  async parseText(text: string, language: string, options: ASTOptions = {}): Promise<Symbol[]> {
```

ESLint

[error] 'options' is assigned a value but never used. Allowed unused args must match /^_/u.

(@typescript-eslint/no-unused-vars)